cli/azd/pkg/infra/azure_resource_manager.go

```diff
@@ -38,17 +38,38 @@ func NewAzureResourceManager(azCli azcli.AzCli) *AzureResourceManager {
 	}
 }
 
+var hasTargetResourceData = func(operation *armresources.DeploymentOperation) bool {
+	return operation.Properties.TargetResource != nil
+}
+
+var isDeploymentType = func(operation *armresources.DeploymentOperation) bool {
+	return *operation.Properties.TargetResource.ResourceType == string(AzureResourceTypeDeployment)
+}
+
+var hasResourceType = func(operation *armresources.DeploymentOperation) bool {
+	return strings.TrimSpace(*operation.Properties.TargetResource.ResourceType) != ""
+}
+
+var isCreateOperation = func(operation *armresources.DeploymentOperation) bool {
+	return *operation.Properties.ProvisioningOperation == armresources.ProvisioningOperationCreate
+}
+
+var hasTimeStamp = func(operation *armresources.DeploymentOperation) bool {
+	return operation.Properties.Timestamp != nil
+}
+
 func (rm *AzureResourceManager) GetDeploymentResourceOperations(
 	ctx context.Context,
 	scope Scope,
 	queryStart *time.Time,
 ) ([]*armresources.DeploymentOperation, error) {
 	// Gets all the scope level resource operations
-	resourceOperations, err := scope.GetResourceOperations(ctx)
+	topLevelDeploymentOperations, err := scope.GetResourceOperations(ctx)
```

Comment:
I think we should filter to deployment operations here? It simplifies handling, avoids creating `topLevel` vs. `allLevel`?

Reply:
Nvm, I can see why you're doing it this way since the contract is to return all non-deployment operations.